homeassistant/components/weheat/__init__.py

```diff
@@ -52,14 +58,30 @@ async def async_setup_entry(hass: HomeAssistant, entry: WeheatConfigEntry) -> bo
     except UnauthorizedException as error:
         raise ConfigEntryAuthFailed from error
 
+    nr_of_pumps = len(discovered_heat_pumps)
+
     for pump_info in discovered_heat_pumps:
         LOGGER.debug("Adding %s", pump_info)
-        # for each pump, add a coordinator
-        new_coordinator = WeheatDataUpdateCoordinator(hass, entry, session, pump_info)
+        # for each pump, add the coordinators
+
+        new_heat_pump = HeatPumpInfo(pump_info)
+        new_data_coordinator = WeheatDataUpdateCoordinator(
+            hass, entry, session, pump_info, nr_of_pumps
+        )
+        new_energy_coordinator = WeheatEnergyUpdateCoordinator(
+            hass, entry, session, pump_info
+        )
 
-        await new_coordinator.async_config_entry_first_refresh()
+        await new_data_coordinator.async_config_entry_first_refresh()
+        await new_energy_coordinator.async_config_entry_first_refresh()
```

Comment:
Should we parallelise these? As in, async.gather?
We have new 3 awaits in a for loop, which is generally not recommended.